neo4j/driver/Neo4jTransaction.java

```diff
@@ -27,29 +27,57 @@
 
 import java.util.List;
 
-public class Neo4jTransaction implements Transaction {
+public abstract class Neo4jTransaction implements Transaction {
 
     private static final Logger LOG = LoggerFactory.getLogger(Neo4jTransaction.class);
 
-    private final Session session;
+    protected final Session session;
 
     public Neo4jTransaction(Session session) {
```

Comment:
Does this constructor need to remain public now it's always used internally?

Reply:
I think you may have forgotten about this comment.

Reply:
Oops missed it, addressed now